src/Symfony/Component/Notifier/Bridge/Sendinblue/SendinblueTransportFactory.php

```diff
@@ -0,0 +1,53 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Notifier\Bridge\Sendinblue;
+
+use Symfony\Component\Notifier\Exception\IncompleteDsnException;
+use Symfony\Component\Notifier\Exception\UnsupportedSchemeException;
+use Symfony\Component\Notifier\Transport\AbstractTransportFactory;
+use Symfony\Component\Notifier\Transport\Dsn;
+use Symfony\Component\Notifier\Transport\TransportInterface;
+
+/**
+ * @author Pierre Tondereau <pierre.tondereau@protonmail.com>
+ *
+ * @experimental in 5.2
+ */
+final class SendinblueTransportFactory extends AbstractTransportFactory
+{
+    /**
+     * @return SendinblueTransport
+     */
+    public function create(Dsn $dsn): TransportInterface
+    {
+        $scheme = $dsn->getScheme();
+        $apiKey = $this->getUser($dsn);
+        $sender = $dsn->getOption('sender');
+        $host = 'default' === $dsn->getHost() ? null : $dsn->getHost();
+        $port = $dsn->getPort();
+
+        if (!$sender) {
+            throw new IncompleteDsnException('Missing sender.', $dsn->getOriginalDsn());
```

Comment:
why not throwing this early toa void useless computation?

Reply:
done